packages/flutter/lib/src/gestures/tap.dart

```diff
@@ -18,12 +18,15 @@ import 'recognizer.dart';
 class TapDownDetails {
   /// Creates details for a [GestureTapDownCallback].
   ///
-  /// The [globalPosition] argument must not be null.
-  TapDownDetails({ this.globalPosition = Offset.zero })
-    : assert(globalPosition != null);
+  /// Arguments [globalPosition] and [buttons] must not be null.
+  TapDownDetails({ this.globalPosition = Offset.zero, this.buttons = 0 })
```

Comment:
I think before we make this change we should update the Android and iOS logic to treat fingers-touching-surface and stylus-touching-surface as button=1.

Reply:
For now the number of set bits is not checked, so a `buttons = 0` will also work as long as it stays 0.

Reply:
Well you're right, for it to work without the change, the callback also needs `device`.

Reply:
> @Hixie The plan is to apply it to long tap, drag and double click later.

Since they share a common superclass, does it make sense to fix them all at once?

Reply:
It's a good idea, but technically the concept of `PrimaryPointerGR` does not necessarily imply "restricted to single button". Do you think I should implement it in `PrimaryPointerGR` or a new intermediate class?